### PR TITLE
fix(#8160): Add infodoc rewrite retries in outbound

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Adds infinite retries in writing infodocs when pushing outbound tasks. 

medic/cht-core#8160

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8160-retry-infodoc-write/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8160-retry-infodoc-write/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8160-retry-infodoc-write/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Hi @jkuester . Would you mind reviewing this? Thanks a lot!

File: sentinel/src/schedule/outbound.js [hunk 1]
```diff
@@ -112,28 +115,14 @@ const singlePush = (taskDoc, medicDoc, infoDoc, config, key) => Promise.resolve(
...
-      });
-  }).catch(() => {
+    return outbound
+      .send(config, key, medicDoc, infoDoc)
+      // Worked, remove entry from queue and store infodoc that outbound service has updated
+      .then(() => removeConfigKeyFromTask(taskDoc, key))
+      .then(() => infodocLib.saveCompletedTasks(medicDoc._id, infoDoc));
+  })
+  .catch((err) => {
+    logger.warn(`Unable to push ${medicDoc._id} for ${key}: %eo`, err);
```
Comment (jkuester) [inline sentinel/src/schedule/outbound.js:125]: Out of curiosity, what is the `%eo` doing here?

Comment (dianabarsan) [reply sentinel/src/schedule/outbound.js:125]: Oh, thanks for noticing, this should just be %o!

Comment (jkuester) [reply sentinel/src/schedule/outbound.js]: :sweat_smile: Here I was thinking this was some special JS thing I did not know about yet...

**Pull request merged**